ambry-router/src/main/java/com.github.ambry.router/GetBlobOperation.java

```diff
@@ -1219,11 +1223,21 @@ private void handleMetadataBlob(BlobData blobData, byte[] userMetadata, ByteBuff
       chunkMetadataList = compositeBlobInfo.getChunkMetadataList();
       boolean rangeResolutionFailure = false;
       try {
+        if (options.getBlobOptions.getBlobSegment() != -1) {
+          long requestedSegment = options.getBlobOptions.getBlobSegment();
+          if (requestedSegment < 0 || requestedSegment >= chunkMetadataList.size()) {
+            throw new IllegalArgumentException(
+                "Bad segment number: " + requestedSegment + ", num of keys: " + chunkMetadataList.size());
+          }
+          chunkMetadataList = chunkMetadataList.subList((int) requestedSegment, (int) requestedSegment + 1);
```

Comment:
All the more reason to make blobSegment an int.

Reply:
fixed